### DISCUSSION TITLE
Support for incremental `TextDocumentSyncKind`

### DISCUSSION THREAD

**Discussion by cperivol** [joaotavora/eglot#1360] [Ideas]
Hi! I noticed the eglot slows down emacs at times and I think I have narrowed it down to it happening when I have a big file open. My theory is that it is sending the entire buffer too often. Does eglot support incremental syncing? If not is anyone working on it? I am willing to take a swing at it if there it's not there and there is no fundamental reason it can't be supported.

**Comment by joaotavora** [comment]
> Does eglot support incremental syncing

Yes.

For incremental syncing to work, you need both server and client to support it.

> My theory is that it is sending the entire buffer too often.

You can confirm theories with experiments :-)

**Reply by joaotavora** [reply]
> > eglot slows down emacs at times and I think 

> You can confirm theories with experiments :-)

... and measurements.  `M-x profiler-start`, `M-x profiler-stop`, `M-x profiler-report` and `emacs -Q` runs to rule out external factors are your friends.
